var/spack/repos/builtin/packages/rocblas/package.py

```diff
@@ -106,6 +106,7 @@ class Rocblas(CMakePackage):
     conflicts("amdgpu_target=gfx1011", when="@:4.2.1")
     conflicts("amdgpu_target=gfx1012", when="@:4.2.1")
     conflicts("amdgpu_target=gfx1030", when="@:4.2.1")
+    conflicts("%gcc@12.2.0", when="@5.2.1:5.2.3")
```

Comment:
```suggestion
    conflicts("%gcc@12", when="@5.2.1:5.2.3")
```
as I'm quite sure this is going to fail with any minor version of GCC 12.

Could you also add a link e.g. to https://reviews.llvm.org/D124866 to explain the conflict?

For context for others, this _should_ be fixed in the upcoming HIP 5.3.0 (based on llvm 15) but hasn't been tested. That's why the upper bound is the restrictive 5.2.3. If it turns out to be needed with 5.3.0 as well we can always change the bound later.

Reply:
To add more context, compiling libstdc++ version 12's `<memory>` header with `hipcc` is what fails to compile. In that sense this could be a blanket conflict between GCC 12 and HIP <= 5.2. However, it seems like rocBLAS doesn't include `<memory>` in every version (at least in the files that are compiled by `hipcc`) so there's also a lower bound on the version in the conflict.

Other projects that attempt to compile `<memory>` with `hipcc` and GCC 12 are likely to hit the same issue though.